### PR TITLE
feat: new onboarding UI

### PR DESCRIPTION
**Pull request by sonle-geekyants**
**- What I did**
Update onboarding UI for these tickets:
- [#321](https://github.com/atsign-foundation/at_widgets/issues/321) - Offer option to upload key file in onboarding widget
- [#352](https://github.com/atsign-foundation/at_widgets/issues/352) - Onboarding Widget - Changes to the @sign activation process
- [#398](https://github.com/atsign-foundation/at_widgets/issues/398) - Wavi Mobile - Some elements are not visible on dark themes (the elements are on on_boarding package)

**- How I did it**
Because the old package architecture was very complex and difficult to maintain so I refactored all of them
- Re-create all screens
- Update screen flow to fit the requirement
- Clean code
After done I requested a tester to test it on AtmospherePro. And now it's stable.

**- How to verify it**
- Just review coding convention and check clean code if need

**- Description for the changelog**

### CODE REVIEW

File: at_onboarding_flutter/example/lib/main.dart [hunk 1]
```diff
@@ -102,36 +105,78 @@ class _MyAppState extends State<MyApp> {
...
+                        switch (result.status) {
+                          case AtOnboardingResultStatus.success:
+                            final atsign = result.atsign;
+                            Navigator.push(
+                                context,
+                                MaterialPageRoute(
+                                    builder: (_) => const HomeScreen()));
+                            break;
+                          case AtOnboardingResultStatus.error:
+                            // TODO: handle onboard failure
```
Comment (nitesh2599) [inline at_onboarding_flutter/example/lib/main.dart]: @sonle-geekyants Can we add an error snackbar here, to show something went wrong while onboarding.

File: at_onboarding_flutter/lib/at_onboarding_result.dart [hunk 2]
```diff
@@ -0,0 +1,45 @@
+enum AtOnboardingResultStatus {
+  success, //Authenticate success
+  error, //Authenticate error
+  cancel, //User canceled
+}
+
+class AtOnboardingResult {
```
Comment (nitesh2599) [inline at_onboarding_flutter/lib/at_onboarding_result.dart:12]: Can we please have one line comments for `AtOnboardingResult` and its members as well.

File: at_onboarding_flutter/lib/screen/at_onboarding_activate_screen.dart [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    try {
+      atsign = atsign.startsWith('@') ? atsign : '@$atsign';
+
+      bool isExist = await _onboardingService.isExistingAtsign(atsign);
+      if (isExist) {
+        await _showAlertDialog(
+            AtOnboardingErrorToString().pairedAtsign(atsign));
+        return;
+      }
+      await Future.delayed(const Duration(seconds: 10));
```
Comment (nitesh2599) [inline at_onboarding_flutter/lib/screen/at_onboarding_activate_screen.dart:174]: @sachins-geekyants @sonle-geekyants Can we add the reason of adding this delay here.

File: at_onboarding_flutter/test/at_onboarding_flutter_test.dart [hunk 4]
```diff
@@ -1,70 +1,70 @@
...
-          findsOneWidget);
-      expect(find.byType(CustomDialog), findsOneWidget);
-      await tester.tap(find.byKey(uniqueKey));
-      expect(_atsign, 'alice💙');
-      await tester.pump();
-      expect(find.byType(CustomDialog), findsOneWidget);
-    });
-  });
-}
+// import 'package:at_onboarding_flutter/services/size_config.dart';
```
Comment (sachins-geekyants) [inline at_onboarding_flutter/test/at_onboarding_flutter_test.dart]: Can we remove the commented code?

File: at_onboarding_flutter/lib/screen/at_onboarding_activate_screen.dart [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+          AtOnboardingStrings.atsignNull,
+        );
+      } else {
+        await showErrorDialog('Response Time out');
+      }
+    } catch (e) {
+      if (e == AtOnboardingResponseStatus.authFailed) {
+        await _showAlertDialog(e, title: 'Auth Failed');
+      } else if (e == AtOnboardingResponseStatus.serverNotReached) {
+        await _processSharedSecret(atsign: atsign, secret: secret);
```
Comment (nitesh2599) [inline at_onboarding_flutter/lib/screen/at_onboarding_activate_screen.dart]: @sonle-geekyants As we are calling the same function from inside of this function, can we run into a scenario where this function keeps on calling itself.

File: at_onboarding_flutter/lib/screen/at_onboarding_input_atsign_screen.dart [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+      _AtOnboardingInputAtSignScreenState();
+}
+
+class _AtOnboardingInputAtSignScreenState
+    extends State<AtOnboardingInputAtSignScreen> {
+  final TextEditingController _atsignController = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
```
Comment (nitesh2599) [inline at_onboarding_flutter/lib/screen/at_onboarding_input_atsign_screen.dart]: Also, if not needed, can we remove these functions.

File: at_onboarding_flutter/lib/screen/at_onboarding_pair_screen.dart [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+      setState(() {});
+      bool status = false;
+      status =
+          await registerPersona(widget.atSign, _emailController.text, context);
+      isParing = false;
+      setState(() {});
+      if (status) {
+        _showOTPScreen();
+      } else {
+        //Todo:
```
Comment (nitesh2599) [inline at_onboarding_flutter/lib/screen/at_onboarding_pair_screen.dart]: Do we need to handle this as well ?

File: at_onboarding_flutter/lib/screen/at_onboarding_qrcode_screen.dart [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+  void onScannedData(String? qrCode) async {
+    if (isDetecting) {
+      return;
+    }
+    isDetecting = true;
+    if ((qrCode ?? '').isNotEmpty) {
+      List<String> values = (qrCode ?? '').split(':');
+      if (values.length == 2) {
+        controller?.pauseCamera();
+        await Future.delayed(const Duration(milliseconds: 400));
```
Comment (nitesh2599) [inline at_onboarding_flutter/lib/screen/at_onboarding_qrcode_screen.dart:120]: Can we add a reason of adding this timer ?

File: at_onboarding_flutter/lib/screen/at_onboarding_reset_screen.dart [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+import 'package:at_onboarding_flutter/utils/at_onboarding_strings.dart';
+import 'package:at_onboarding_flutter/widgets/at_onboarding_button.dart';
+import 'package:at_onboarding_flutter/widgets/at_onboarding_dialog.dart';
+import 'package:flutter/material.dart';
+
+import '../services/at_onboarding_config.dart';
+import '../services/sdk_service.dart';
+
+enum AtOnboardingResetResult {
+  canceled,
```
Comment (nitesh2599) [inline at_onboarding_flutter/lib/screen/at_onboarding_reset_screen.dart]: Can we change it to `cancelled` .

File: at_onboarding_flutter/lib/screen/at_onboarding_qrcode_screen.dart (hunk 8 above)
Comment (sonle-geekyants) [reply at_onboarding_flutter/lib/screen/at_onboarding_qrcode_screen.dart:120]: Added the comment.
"//It's issue from camera library so we need to add a delay to waiting for pause camera"

File: at_onboarding_flutter/lib/screen/at_onboarding_activate_screen.dart (hunk 3 above)
Comment (sonle-geekyants) [reply at_onboarding_flutter/lib/screen/at_onboarding_activate_screen.dart:174]: Added comment: 
//Delay 10s for waiting for ServerStatus change to teapot when activating an atsign

Sachin also talks about this issue. Don't have any else solution for now.

File: at_onboarding_flutter/lib/at_onboarding_result.dart (hunk 2 above)
Comment (sonle-geekyants) [reply at_onboarding_flutter/lib/at_onboarding_result.dart:12]: Fixed

**Review by nitesh2599 — APPROVED** [review]

**Review by sachins-geekyants — APPROVED** [review]

**Pull request merged**